src/bot/actions.js

```diff
@@ -0,0 +1,123 @@
+
+let ax = axios.create({
+  baseURL: process.env.BACKEND_URL || 'http://127.0.0.1:5555',
+  timeout: 10000,
+});
+
+const isValidAddress = (address) => {
+  try {
+    keyring.decodeAddress(address);
+  } catch (error) {
+    return false;
+  }
+  return true;
+}
+
+const requestDrip = ({
+  sender,
+  address,
+  amount,
+  dripType,
+}) => ax.post('/drip', {
+  sender,
+  address,
+  amount,
+  dripType,
+});
+
+const drip = async (sender, address) => {
+	if (_.isEmpty(address)) {
+    return 'Please enter a wallet address after !drip.';
+  }
+
+  if (!isValidAddress(address)) {
+    return `The address ${address} entered is incompatible to ${networkName}.`;
+  }
+
+  const res = await requestDrip({
+    sender,
+    address,
+    amount: sendAmount * units,
+    dripType: DRIP_TYPE.NORMAL,
+  });
+
+  if (res.data === 'LIMIT') {
+    return `Your Discord ID or the address has reached its daily quota. Please request only once every 24 hours.`;
+  }
+  return `I just sent ${sendAmount} ${tokenSymbol} to address ${address}. Extrinsic hash: ${res.data.hash}.`;
+}
+
+const parseTime = (time) => {
+  try {
+  const matches = /^(\d+)\/(\d+) at (\d+)(am|pm)/s.exec(time);
+  if (!matches) {
+    return null;
+  }
+
+  const month = Number(matches[1]);
+  const day = Number(matches[2]);
+  const noon = matches[4];
+  let hour = Number(matches[3]);
+  hour = noon === 'pm' ? hour + 12 : hour;
+  
+  const now = moment();
+  const timeStr = `${now.year()}-${_.padStart(month, 2, '0')}-${_.padStart(day, 2, '0')}T${_.padStart(hour, 2, '0')}:00:00.000Z`;
+
+  const newTime = moment(timeStr);
+  if (!newTime.isValid()) {
+    return null;
+  }
+  
+  return newTime;
+  } catch (error) {
+    return null;
+  }
+}
+
+const dripLater = async (sender, address, time) => {
+	if (_.isEmpty(address)) {
+    return 'please enter a wallet address after !drip-later.';
+  }
+
+  if (!isValidAddress(address)) {
+    return `The address ${address} entered is incompatible to ${networkName}.`;
+  }
+
+  const dripTime = parseTime(time);
+  if (!dripTime) {
+    return `Please enter the specified time format(UTC). Example: !drip-later address <4/14 at 5pm>`;
+  }
+  if (dripTime.isBefore(moment())) {
+    return "The time in UTC must be in the future.";
+  }
+  if (dripTime.isAfter(moment().add(maxScheduleSeconds, 'seconds'))) {
+    return `The time in UTC cannot be farther than ${moment.duration(maxScheduleSeconds, 'seconds').asDays()} days.`;
+  }
+
+  const res = await requestDrip({
+    sender,
+    address,
+    amount: sendAmount * units,
+    dripType: DRIP_TYPE.LATER,
+    dripTime: dripTime.valueOf() / 1000,
+  });
+
+  if (res.data === 'LIMIT') {
+    return `Your Discord ID or the address has reached its daily quota. Please request 3 times every 24 hours.`;
+  }
+
+  const { data: { hash, providerId } } = res;
+  return `I will send ${sendAmount} NEU to address ${address} at ${time}. Extrinsic hash: ${hash}. Your provided_id: ${providerId}.`;
```

Comment:
Add timezone to this: `I will send ${sendAmount} NEU to address ${address} at ${time} UTC.`

Reply:
Fixed